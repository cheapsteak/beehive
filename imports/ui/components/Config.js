// Framework
import React from 'react';

// Components
import { Container, Row, Col } from 'reactstrap';
import FieldRows from './FieldRows';

/**
 * A list of Configuration options
 *
 * @param {Object} props All the properties passed to the React Component.
 * @returns {Element} Stateless functional React component.
 */
const Config = ({ configFields = {}, configSettingAction }) => (
  <aside>
    <Container className="config" fluid>
      <Row>
        <Col>
          <FieldRows
            heading="Configuration"
            fields={configFields}
            stateSettingAction={configSettingAction}
          />
        </Col>
      </Row>
    </Container>
  </aside>
);

export default Config;
