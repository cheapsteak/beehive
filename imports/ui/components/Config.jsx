// Framework
import React from "react";

// Components
import { Container, Row, Col } from "reactstrap";

/**
 * A list of Configuration options
 *
 * @param {Object} props All the properties passed to the React Component.
 * @returns {Element} Stateless functional React component.
 */
const Config = () =>
  <Container className="config" fluid>
    <Row>
      <Col>
        <h3>Set Jobs</h3>
        <p>Job A - $20/hour</p>
        <p>Job B - $30/hour</p>
      </Col>
    </Row>
  </Container>;

export default Config;
