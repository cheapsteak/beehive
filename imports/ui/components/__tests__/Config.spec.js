import React from "react";
import ReactTestRenderer from "react-test-renderer";
import { shallow } from "enzyme";

import Config from "../Config";

describe("<Config />", () => {
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      configFields: { tickRate: 50 },
      configSettingAction: jest.fn()
    };
  });

  describe("Config SnapShot", () => {
    it("renders config correctly with default props", () => {
      delete defaultProps.configFields;
      const tree = ReactTestRenderer.create(
        <Config {...defaultProps} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("renders config correctly", () => {
      const tree = ReactTestRenderer.create(<Config />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("Config Props", () => {
    let wrapper;

    beforeEach(() => {
      const wrapper = shallow(<Config {...defaultProps} />);
    });
  });
});
