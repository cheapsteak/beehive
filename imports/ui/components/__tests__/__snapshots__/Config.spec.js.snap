// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Config /> Config SnapShot renders config correctly 1`] = `
<aside>
  <div
    className="config container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="field-rows px-0 container-fluid"
        >
          <div
            className="no-gutters row"
          >
            <div
              className="col"
            >
              <h3>
                Configuration
              </h3>
            </div>
          </div>
          <div
            className="no-gutters row"
          >
            <div
              className="col"
            >
              <form
                className=""
              >
                <div
                  className="field-row form-group"
                >
                  <label
                    className=""
                    htmlFor="tickRate-number-field-row-input"
                  >
                    Tick Rate
                  </label>
                  <input
                    className="form-control"
                    id="tickRate-number-field-row-input"
                    name="tickRate-number-field-row-input"
                    onChange={[Function]}
                    placeholder="Tick Rate"
                    type="text"
                    value={50}
                  />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</aside>
`;

exports[`<Config /> Config SnapShot renders config correctly with default props 1`] = `
<aside>
  <div
    className="config container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="field-rows px-0 container-fluid"
        >
          <div
            className="no-gutters row"
          >
            <div
              className="col"
            >
              <h3>
                Configuration
              </h3>
            </div>
          </div>
          <div
            className="no-gutters row"
          >
            <div
              className="col"
            >
              <form
                className=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</aside>
`;
